distributed/scheduler.py

```diff
@@ -7878,7 +7878,11 @@ def _exit_processing_common(
     state.release_resources(ts, ws)
 
     # If a slot has opened up for a queued task, schedule it.
-    if state.queued and not _worker_full(ws, state.WORKER_SATURATION):
+    if (
+        state.queued
+        and ws.status == Status.running
```

Comment:
This is true right now but if we ever had more `Status` types, of which some might actually be runnable, this could cause a deadlock, couldn't it?
I know this is pretty hypothetical but my point is that we do not change the behavior but encode more assumptions in the code for a perf improvement you call yourself minuscule (and rare)

I don't mean to block this if you consider it truly better readable but I'm a soft -0

Reply:
That seems like a pretty far-fetched problem. There are already 18 instances of `== Status.running` in `scheduler.py`.

I've added a test. This is still not a big deal, it's just a small tidying up.